### ISSUE TITLE
Unable to build

### ISSUE THREAD

**Issue by cjuniorfox** [Fenrir-ode/webserver#1]
Hi, I'm trying to build the package with no success,
I'm installed "cmake", "" build-essential" and cloned the repository "https://github.com/cesanta/mongoose.git" into "mongoose" folder. I  followed the instructions, but unfortunately, when I "make" the project, the build hangs into an error. It's there anything I can do?

The error follows:

Best Regards

```
$ make
[  5%] Performing update step for 'libchdr'
Current branch master is up to date.
[ 11%] Performing configure step for 'libchdr'
-- Configuring done
-- Generating done
-- Build files have been written to: /home/junior/projects/webserver/build/libchdr/src/libchdr-build
[ 16%] Performing build step for 'libchdr'
[ 32%] Built target zlib
[ 62%] Built target lzma
[ 78%] Built target chdr
[ 94%] Built target chdr-static
[100%] Built target chdr-benchmark
[ 22%] Performing install step for 'libchdr'
Skipping install step.
[ 27%] Completed 'libchdr'
[ 44%] Built target libchdr
[ 50%] Building C object CMakeFiles/FenrirServer.dir/src/menu.http.c.o
/home/junior/projects/webserver/src/menu.http.c: In function ‘_printdirentry’:
/home/junior/projects/webserver/src/menu.http.c:52:25: error: ‘struct mg_fs’ has no member named ‘stat’; did you mean ‘st’?
   52 |     flags = mg_fs_posix.stat(path, &size, &mtime);
      |                         ^~~~
      |                         st
/home/junior/projects/webserver/src/menu.http.c: In function ‘ftw_process’:
/home/junior/projects/webserver/src/menu.http.c:94:46: warning: passing argument 1 of ‘__xpg_basename’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   94 |             const char *entryname = basename(name);
      |                                              ^~~~
In file included from /home/junior/projects/webserver/src/menu.http.c:4:
/usr/include/libgen.h:34:14: note: expected ‘char *’ but argument is of type ‘const char *’
   34 | extern char *__xpg_basename (char *__path) __THROW;
      |              ^~~~~~~~~~~~~~
/home/junior/projects/webserver/src/menu.http.c: In function ‘_printdirentry’:
/home/junior/projects/webserver/src/menu.http.c:51:41: warning: ‘snprintf’ output may be truncated before the last format character [-Wformat-truncation=]
   51 |     snprintf(path, sizeof(path), "%s%c%s", fud->image_path, '/', name);
      |                                         ^
/home/junior/projects/webserver/src/menu.http.c:51:5: note: ‘snprintf’ output 2 or more bytes (assuming 4097) into a destination of size 4096
   51 |     snprintf(path, sizeof(path), "%s%c%s", fud->image_path, '/', name);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/FenrirServer.dir/build.make:141: CMakeFiles/FenrirServer.dir/src/menu.http.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:78: CMakeFiles/FenrirServer.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

**Comment by gnoling** [comment]
How did you clone mongoose?

Testing the current repository this way compiled for me:

```
git clone --recurse-submodules  https://github.com/Fenrir-ode/webserver.git test-fenrir-build
cd test-fenrir-build/
mkdir build
cd build
cmake ..
make
```

My first thought is if you cloned it directly it wouldn't be at the correct commit/version for mongoose.

**Comment by cjuniorfox** [comment]
Your thought makes sense. I clonned again with the "recursive" flag and I was able to properly build the package.
My advice it's add those instructions at README file for the repository.

By the way, it's there any place to post any feedback reporting issues and suggestions for the Fernir ODE? I made a bunch of testes with, and I would like to report issues and suggestions about the functionalities and about the documentation.

Regards.
